local_db/api.py

```diff
@@ -0,0 +1,64 @@
+from dataclasses import dataclass
+
+from airlock.api import ProviderAPI, ReleaseRequest, User
+from local_db.models import RequestMetadata
+
+
+@dataclass
+class LocalDBProvider(ProviderAPI):
+    """Implementation of ProviderAPI using local_db models to store data."""
+
+    def _request(self, metadata: RequestMetadata = None):
+        """Unpack the db data into the Request object."""
+        return ReleaseRequest(
+            id=metadata.id,
+            workspace=metadata.workspace,
+            author=metadata.author,
+            created_at=metadata.created_at,
+        )
+
+    def _create_release_request(self, **kwargs):
+        metadata = RequestMetadata.objects.create(**kwargs)
+        return self._request(metadata)
+
+    def _find_metadata(self, request_id: str):
+        try:
+            return RequestMetadata.objects.get(id=request_id)
+        except RequestMetadata.DoesNotExist:
+            raise self.ReleaseRequestNotFound(request_id)
+
+    def get_release_request(self, request_id: str):
+        return self._request(self._find_metadata(request_id))
+
+    def get_current_request(self, workspace: str, user: User, create=False):
+        requests = list(
+            RequestMetadata.objects.filter(
+                workspace=workspace,
+                author=user.username,
+            )
+        )
+        n = len(requests)
+        if n > 1:
+            raise Exception(
+                f"Multiple active release requests for user {user.username} in workspace {workspace}"
+            )
+        elif n == 1:
+            return self._request(requests[0])
+        elif create:
+            return self._create_release_request(
+                workspace=workspace,
+                author=user.username,
+            )
+
+    def get_requests_for_user(self, user: User):
```

Comment:
It feels a little risky to have the same method used for getting requests in the two different user contexts.

Reply:
Yes - this will have to change in future. But again, this was part of previous change - I am just implementing it here.

In the UI, I think we will want to separate "My Requests" (ones you've authored) from "Requests to review" (output checker only). When we do, we can have different urls/views/API functions?